### PR TITLE
Extend DifferentialOperator to vector valued functions

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: sumpy/expansion/diff_op.py [hunk 1]
```diff
@@ -32,69 +33,147 @@
...
-        return self + (-1)*other_pde
+    def __sub__(self, other_diff_op):
+        return self + (-1)*other_diff_op
 
     def __repr__(self):
-        return f"DifferentialOperator({self.dim}, {repr(self.mi_to_coeff)})"
+        return f"DifferentialOperator({self.dim}, {repr(self.eqs)})"
+
+    def __getitem__(self, idx):
+        item = list(self.eqs).__getitem__(idx)
```
Comment (inducer) [inline sumpy/expansion/diff_op.py:89]: Why the `list`?

File: sumpy/expansion/diff_op.py [hunk 2]
```diff
@@ -32,69 +33,147 @@
...
 
 class DifferentialOperator(object):
     r"""
-    Represents a scalar, constant-coefficient DifferentialOperator of
-    dimension `dim`. It is represented by a frozen dictionary.
-    The dictionary maps a multi-index given as a tuple to the coefficient.
-    This object is immutable.
+    Represents a constant-coefficient DifferentialOperator of a vector-valued
+    function with `dim` variables. It is represented by a tuple of
+    frozen dictionaries. The dictionary maps a (multi-index, vector index)
```
Comment (inducer) [inline sumpy/expansion/diff_op.py:43]: Make use of `DerivativeIdentifier` here?

File: sumpy/expansion/diff_op.py [hunk 3]
```diff
@@ -32,69 +33,147 @@
 .. autoclass:: DifferentialOperator
 """
 
+DerivativeIdentifier = namedtuple("DerivativeIdentifier", ["mi", "vec_idx"])
```
Comment (inducer) [inline sumpy/expansion/diff_op.py:36]: Add this to the docs?

File: sumpy/expansion/diff_op.py (hunk 2 above)
Comment (inducer) [inline sumpy/expansion/diff_op.py:43]: What is a frozen dictionary?

File: sumpy/expansion/diff_op.py [hunk 4]
```diff
@@ -32,69 +33,147 @@
...
+    function with `dim` variables. It is represented by a tuple of
+    frozen dictionaries. The dictionary maps a (multi-index, vector index)
+    tuple to the coefficient. This object is immutable.
     """
-    def __init__(self, dim, mi_to_coeff):
+    def __init__(self, dim, *eqs):
         """
         :arg dim: dimension of the DifferentialOperator
-        :arg mi_to_coeff: A dictionary mapping a multi-index to a coefficient
+        :arg eqs: A list of dictionaries mapping a (multi-index, vector-index)
```
Comment (inducer) [inline sumpy/expansion/diff_op.py:49]: `DerivativeIdentifier`?

File: sumpy/expansion/diff_op.py [hunk 5]
```diff
@@ -32,69 +33,147 @@
 .. autoclass:: DifferentialOperator
 """
 
+DerivativeIdentifier = namedtuple("DerivativeIdentifier", ["mi", "vec_idx"])
+
 
 class DifferentialOperator(object):
```
Comment (inducer) [inline sumpy/expansion/diff_op.py:39]:
```suggestion
class LinearPDESystemOperator(object):
```

**Review by inducer — COMMENTED** [review]
Thanks for working on this!

File: sumpy/expansion/diff_op.py (hunk 3 above)
Comment (isuruf) [reply sumpy/expansion/diff_op.py:38]: Done

File: sumpy/expansion/diff_op.py (hunk 2 above)
Comment (isuruf) [reply sumpy/expansion/diff_op.py]: Changed to immutable dictionary

**Comment by isuruf** [comment]
Thanks for the review. I've made the changes

**Comment by inducer** [comment]
LGTM. Thanks!

**Pull request merged**